packages/ember-data/lib/system/model/errors.js

```diff
@@ -297,6 +284,7 @@ export default Ember.Object.extend(Ember.Enumerable, Ember.Evented, {
 
     var content = get(this, 'content').rejectBy('attribute', attribute);
     get(this, 'content').setObjects(content);
+    get(this, 'errorsByAttributeName').remove(attribute);
```

Comment:
This is because currently the `RAISE_ON_DEPRECATION` env is not set to `true`, which is tracked in #3233